geom.go

```diff
@@ -42,22 +46,24 @@ type MultiLineStringer interface {
 	LineStrings() [][][2]float64
 }
 
+// MultiLineStringSetter is a mutable MultiLineStringer.
 type MultiLineStringSetter interface {
 	MultiLineStringer
 	SetLineStrings([][][2]float64) error
 }
 
-// 	Polygoner is a geometry consisting of multiple closed LineStrings.
-//	There must be only one exterior LineString with a clockwise winding order.
-//	There may be one or more interior LineStrings with a counterclockwise winding orders.
+// Polygoner is a geometry consisting of multiple closed LineStrings.
+// There must be only one exterior LineString with a clockwise winding order.
+// There may be one or more interior LineStrings with a counterclockwise winding orders.
 type Polygoner interface {
 	Geometry
-	SubLineStrings() [][][2]float64
```

Comment:
That is not a safe way of meeting this need - you can't stop someone from implementing interfaces. I submit that if a library can't handle different geometry types then it is broken. I recommend adding a dimensionality and possibly cardinality functions to the geometry interface. Then you can definitively disambiguate types and still allow for extensibility.

Reply:
Right, that's why they are different interfaces. If a developer wants their type to behave like a multi-linestring or a polygon they can implement both interfaces. Right now, you are making it so that all multi-linestrings and polygons behave the same. In addition to removing the ability for the compiler to catch when some passes in something that should not be considered a polygon v.s. a multi-linestring.

We want to keep the interfaces as small as possible. I don't think adding more to the geometry interface is better than what we have currently. That just add's additional runtime checks one has to do to validate the things they are receiving.